contrib/format-esri/src/main/java/org/apache/drill/exec/store/esri/ShpBatchReader.java

```diff
@@ -125,7 +125,7 @@ public boolean next() {
 
   private void openFile(FileSchemaNegotiator negotiator) {
     try {
-      fileReaderShp = negotiator.fileSystem().open(split.getPath());
+      fileReaderShp = negotiator.fileSystem().openDecompressedInputStream(split.getPath());
```

Comment:
I have some doubts regarding the safety of the `openDecompressedInputStream()` method. It contains the following line:
```
byte[] bytes = IOUtils.toByteArray(compressedStream);
```
that will read all the stream to the memory, but it may cause OOM.
Why we can't use here another `openPossiblyCompressedStream()` method?

Reply:
@vvysotskyi 
Thanks for your comment.  The issue that was happening on S3 was that the `inputstream` that gets passed to the format plugin seemed to be compressed in some way, even if the file itself was not compressed. 
This didn't seem to matter for text based file formats, but for formats that read binary data, such as `pcap`, the stream wasn't being decompressed and the result was that Drill couldn't parse the file. 

The `openPossiblyCompressedStream()` method didn't really solve this issue, because you could end up with a compressed stream that the format plugins couldn't read.  I thought another approach would be to put this logic in the format plugins themselves, but I couldn't figure out a way to determine whether the stream was compressed or not after you call the `openPossiblyCompressedStream()`. 

Do you have any suggestions as to how to fix so that we can avoid the OOM?

Reply:
Both methods `openPossiblyCompressedStream()` and `openDecompressedInputStream()` have similar logic in determining compression codec (based on specified path) and producing decompression. The difference is only that the last one places whole the stream to the buffer and wraps it to the `ByteArrayInputStream`. So looks like if `openDecompressedInputStream()` will not help, the `openPossiblyCompressedStream()` also wouldn't work.

If for some reason files from s3 are read as compressed, and there is no way for obtaining info for the compression method when reading the file, we can add a property to the file storage plugin `config` block that will hold the default compression type if any.